braidpool_spec.md

```diff
@@ -544,11 +830,34 @@ A private arrangement can be created where one party *buys future shares* from
 another in exchange for bitcoin up front. This is a *futures* contract, where
 the counterparty to the miner is taking on pool "luck" risk and fee rate risk.
 
+[kp]
+
+Bitcoin can do better. We can build cryptographically secure futures
+contracts. If we can, I think we should.
+
+[/kp]
+
 In order to form hashrate derivatives, it must be posible to deliver shares
 across two different difficulty adjustment windows. Shares in one difficulty
 adjustment window have a different value compared to shares in another window,
 due to the difficulty adjustment itself. If one can compute the derivative
 
+[kp]
+
+The shares from different difficulty windows don't need to be traded. What is
+traded is the fact that shares generated certain amount of bitcoin and that
+reward is what is exchanged.
+
+So miner buys a contract to deliver all the bitcoin they will earn on braidpool
+to market maker, say it will be G BTC. Market maker promises to give F BTC for
+that G BTC. The relationship between F and G is the risk the market maker lives
+on.
+
+The reason is also that market makers have no use for the braidpool shares. They
+have no market value at all. Meanwhile G BTC is G BTC.
+
+[/kp]
+
```

Comment:
The hashrate derivative is a bet on this epoch's shares vs next epoch's shares. (e.g. does the hashrate go up or down?) One can do that by promising to buy the next epoch's shares using this epoch's shares at a fixed price. Could also be done using bitcoin as the denominator. But fundamentally hashrate derivatives are a 2-epoch discrete derivative.